fivegmag_5GMSdAwareApplication/app/src/main/java/com/fivegmag/a5gmsdawareapplication/MainActivity.kt

```diff
@@ -184,10 +191,25 @@ class MainActivity : AppCompatActivity(), AdapterView.OnItemSelectedListener {
                 Manifest.permission.ACCESS_FINE_LOCATION
             ) != PackageManager.PERMISSION_GRANTED
         ) {
-            requestPermissionLauncher.launch(
-                Manifest.permission.ACCESS_FINE_LOCATION
-            )
-        } else {
+            permissions[idx++] = Manifest.permission.ACCESS_FINE_LOCATION
+            needReqPermission = true
+        }
+
+        if (ActivityCompat.checkSelfPermission(
+                this,
+                Manifest.permission.READ_PHONE_NUMBERS
+            ) != PackageManager.PERMISSION_GRANTED
+        ) {
+            permissions[idx] = Manifest.permission.READ_PHONE_NUMBERS
+            needReqPermission = true
+        }
+
+        if (needReqPermission)
```

Comment:
Check if length of `permissions` array is > 0. No need to define `needReqPermission`